mdal/frmts/mdal_2dm.cpp

```diff
@@ -286,3 +286,63 @@ std::unique_ptr<MDAL::Mesh> MDAL::Driver2dm::load( const std::string &meshFile,
 
   return std::unique_ptr<Mesh>( mesh.release() );
 }
+
+void MDAL::Driver2dm::save( const std::string &uri, MDAL_Status *status, MDAL::Mesh *mesh )
+{
+  if ( status ) *status = MDAL_Status::None;
+
+  std::ofstream file( uri, std::ofstream::out );
+
+  if ( !file.is_open() )
+  {
+    if ( status ) *status = MDAL_Status::Err_FailToWriteToDisk;
+  }
+
+  std::string line = "MESH2D";
+  file << line << std::endl;
+
+  //write vertices
+  std::unique_ptr<MDAL::MeshVertexIterator> vertexIterator = mesh->readVertices();
+  double vertex[3];
+  for ( size_t i = 0; i < mesh->verticesCount(); ++i )
+  {
+    vertexIterator->next( 1, vertex );
+    line = "ND ";
+    line.append( std::to_string( i + 1 ) );
+    for ( size_t j = 0; j < 3; ++j )
+    {
+      line.append( " " );
+      line.append( std::to_string( vertex[j] ) );
+    }
+    file << line << std::endl;
+  }
+
+  //write faces
+  std::unique_ptr<MDAL::MeshFaceIterator> faceIterator = mesh->readFaces();
+  for ( size_t i = 0; i < mesh->facesCount(); ++i )
+  {
+    int faceOffsets[1];
+    int vertexIndices[4]; //max 4 vertices for a face
```

Comment:
face can have even more than 4 vertices, check function https://github.com/lutraconsulting/MDAL/blob/b8f79d8e4969b1cfc8641257f4bdba28d2c1937a/mdal/api/mdal.h#L141 `MDAL_M_faceVerticesMaximumCount` .If 2dm driver can only store triangles or quads, best to return error code here.

Reply:
In the last commit, I have added a method in the MDAL::Driver class to return the maximum vertices per face the driver handles. Default is big, and Diver2DM return 4.
So, now we can test if the mesh is compatible with the driver before saving.